tractor/_portal.py

```diff
@@ -427,9 +434,14 @@ async def open_context(
                 log.debug(f'Context {ctx} cancelled gracefully')
 
         except (
-            trio.Cancelled,
-            trio.MultiError,
-            Exception,
+            BaseException,
```

Comment:
This is the main change that gets us the fix.
The alternative would be to include the kbi in the old error set below but it's likely that any other base exception should trigger this same cancellation behavior afaik.